apps/api/src/app/tenant/usecases/update-tenant/update-tenant.usecase.ts

```diff
@@ -0,0 +1,79 @@
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { TenantRepository } from '@novu/dal';
+import { UpdateTenantCommand } from './update-tenant.command';
+import { TenantEntity } from '@novu/dal';
+
+@Injectable()
+export class UpdateTenant {
+  constructor(private tenantRepository: TenantRepository) {}
+
+  async execute(command: UpdateTenantCommand): Promise<TenantEntity> {
+    const tenant = await this.tenantRepository.findOne({
+      _environmentId: command.environmentId,
+      identifier: command.identifier,
+    });
+
+    if (!tenant) {
+      throw new BadRequestException(
+        `Tenant with identifier: ${command.identifier} is not exists under environment ${command.environmentId}`
+      );
+    }
+
+    const updatePayload: Partial<TenantEntity> = {};
+
+    if (command.name != null) {
+      updatePayload.name = command.name;
+    }
+
+    if (command.data != null) {
+      updatePayload.data = command.data;
+    }
+
+    if (command.newIdentifier != null && command.newIdentifier !== command.identifier) {
```

Comment:
```suggestion
    if (command?.newIdentifier !== command.identifier) {
```

Reply:
here we need to make sure that `command.newIdentifier` is not undefined as well, and then check if they are not equal, because of the following case:
`command?.newIdentifier !== command.identifier` could be  `undefine !== 'current_identifier'` and the if will the true but i am not sure we will want to update it with `undefined`

Reply:
This TS statement `command?.newIdentifier !== command.identifier` is compiled to JS like `command?.newIdentifier && command.newIdentifier !== command.identifier`.
So for the `undefined` case it will be like `undefined && command.newIdentifier !== command.identifier`, in the result `false`.

Reply:
Added a dedicated test to check the null/undefined updates just in case.